The job **`mingw-check-tidy`** failed! Check out the build log: [(web)](https://github.com/rust-lang/rust/actions/runs/9964856736/job/27534035565) [(plain)](https://github.com/rust-lang/rust/commit/e8440a86b80e39bfbd9e48d660d0f7c83eccbcc9/checks/27534035565/logs)

<details><summary><i>Click to see the possible cause of the failure (guessed by this bot)</i></summary>

```plain
Getting action download info
Download action repository 'msys2/setup-msys2@v2.22.0' (SHA:cc11e9188b693c2b100158c3322424c4cc1dadea)
Download action repository 'actions/checkout@v4' (SHA:692973e3d937129bcbf40652eb9f2f61becf3332)
Download action repository 'actions/upload-artifact@v4' (SHA:0b2256b8c012f0828dc542b3febcab082c67f72b)
Complete job name: PR - mingw-check-tidy
git config --global core.autocrlf false
shell: /usr/bin/bash --noprofile --norc -e -o pipefail {0}
---
COPY scripts/sccache.sh /scripts/
RUN sh /scripts/sccache.sh

COPY host-x86_64/mingw-check/reuse-requirements.txt /tmp/
RUN pip3 install --no-deps --no-cache-dir --require-hashes -r /tmp/reuse-requirements.txt \
    && pip3 install virtualenv
COPY host-x86_64/mingw-check/validate-toolstate.sh /scripts/
COPY host-x86_64/mingw-check/validate-error-codes.sh /scripts/

# NOTE: intentionally uses python2 for x.py so we can test it still works.
# NOTE: intentionally uses python2 for x.py so we can test it still works.
# validate-toolstate only runs in our CI, so it's ok for it to only support python3.
ENV SCRIPT TIDY_PRINT_DIFF=1 python2.7 ../x.py test \
           --stage 0 src/tools/tidy tidyselftest --extra-checks=py:lint,cpp:fmt
# This file is autogenerated by pip-compile with Python 3.10
# by the following command:
#
#    pip-compile --allow-unsafe --generate-hashes reuse-requirements.in
---

#12 [5/8] COPY host-x86_64/mingw-check/reuse-requirements.txt /tmp/
#12 DONE 0.0s

#13 [6/8] RUN pip3 install --no-deps --no-cache-dir --require-hashes -r /tmp/reuse-requirements.txt     && pip3 install virtualenv
#13 0.446   Downloading binaryornot-0.4.4-py2.py3-none-any.whl (9.0 kB)
#13 0.459 Collecting boolean-py==4.0
#13 0.462   Downloading boolean.py-4.0-py3-none-any.whl (25 kB)
#13 0.473 Collecting chardet==5.1.0
---
#13 4.789 WARNING: Running pip as the 'root' user can result in broken permissions and conflicting behaviour with the system package manager. It is recommended to use a virtual environment instead: https://pip.pypa.io/warnings/venv
#13 5.344 Collecting virtualenv
#13 5.386   Downloading virtualenv-20.26.3-py3-none-any.whl (5.7 MB)
#13 5.471      ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 5.7/5.7 MB 69.5 MB/s eta 0:00:00
#13 5.523 Collecting platformdirs<5,>=3.9.1
#13 5.532   Downloading platformdirs-4.2.2-py3-none-any.whl (18 kB)
#13 5.566 Collecting filelock<4,>=3.12.2
#13 5.574   Downloading filelock-3.15.4-py3-none-any.whl (16 kB)
#13 5.594 Collecting distlib<1,>=0.3.7
#13 5.600   Downloading distlib-0.3.8-py2.py3-none-any.whl (468 kB)
#13 5.608      ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 468.9/468.9 KB 83.6 MB/s eta 0:00:00
#13 5.697 Installing collected packages: distlib, platformdirs, filelock, virtualenv
#13 5.885 Successfully installed distlib-0.3.8 filelock-3.15.4 platformdirs-4.2.2 virtualenv-20.26.3
#13 DONE 6.0s

#14 [7/8] COPY host-x86_64/mingw-check/validate-toolstate.sh /scripts/
#14 DONE 0.0s
---
DirectMap4k:      198592 kB
DirectMap2M:     7141376 kB
DirectMap1G:    11534336 kB
##[endgroup]
Executing TIDY_PRINT_DIFF=1 python2.7 ../x.py test            --stage 0 src/tools/tidy tidyselftest --extra-checks=py:lint,cpp:fmt
+ TIDY_PRINT_DIFF=1 python2.7 ../x.py test --stage 0 src/tools/tidy tidyselftest --extra-checks=py:lint,cpp:fmt
    Finished `dev` profile [unoptimized] target(s) in 0.04s
##[endgroup]
downloading https://ci-artifacts.rust-lang.org/rustc-builds-alt/032be6f7bbe091c7dfa29f115e94b9cc9bae1758/rust-dev-nightly-x86_64-unknown-linux-gnu.tar.xz
extracting /checkout/obj/build/cache/llvm-032be6f7bbe091c7dfa29f115e94b9cc9bae1758-true/rust-dev-nightly-x86_64-unknown-linux-gnu.tar.xz to /checkout/obj/build/x86_64-unknown-linux-gnu/ci-llvm
---
   Compiling tidy v0.1.0 (/checkout/src/tools/tidy)
    Finished `release` profile [optimized] target(s) in 30.22s
##[endgroup]
fmt check
##[error]Diff in /checkout/library/std/src/sys/pal/zkvm/alloc.rs at line 5:
 unsafe impl GlobalAlloc for System {
     #[inline]
     unsafe fn alloc(&self, layout: Layout) -> *mut u8 {
-        unsafe{
-            abi::sys_alloc_aligned(layout.size(), layout.align())
+        unsafe { abi::sys_alloc_aligned(layout.size(), layout.align()) }
     }
 
     #[inline]
     #[inline]
fmt error: Running `"/checkout/obj/build/x86_64-unknown-linux-gnu/rustfmt/bin/rustfmt" "--config-path" "/checkout" "--edition" "2021" "--unstable-features" "--skip-children" "--check" "/checkout/library/std/src/sys/pal/unix/os/tests.rs" "/checkout/library/std/src/sys/pal/unix/thread.rs" "/checkout/library/std/src/sys/pal/unix/fd/tests.rs" "/checkout/library/std/src/sys/pal/unix/fd.rs" "/checkout/library/std/src/sys/pal/unix/kernel_copy.rs" "/checkout/library/std/src/sys/pal/unix/io.rs" "/checkout/library/std/src/sys/pal/unix/args.rs" "/checkout/library/std/src/sys/pal/unix/mod.rs" "/checkout/library/std/src/sys/pal/unix/env.rs" "/checkout/library/std/src/sys/pal/unix/fs/tests.rs" "/checkout/library/std/src/sys/pal/zkvm/os.rs" "/checkout/library/std/src/sys/pal/zkvm/stdio.rs" "/checkout/library/std/src/sys/pal/zkvm/alloc.rs" "/checkout/library/std/src/sys/pal/zkvm/abi.rs" "/checkout/library/std/src/sys/pal/zkvm/args.rs" "/checkout/library/std/src/sys/pal/zkvm/mod.rs" "/checkout/library/std/src/sys/pal/zkvm/env.rs" "/checkout/library/std/src/sys/pal/hermit/os.rs" "/checkout/library/std/src/sys/pal/hermit/stdio.rs" "/checkout/library/std/src/sys/pal/hermit/time.rs" "/checkout/library/std/src/sys/pal/hermit/net.rs" "/checkout/library/std/src/sys/pal/hermit/futex.rs" "/checkout/library/std/src/sys/pal/hermit/fs.rs" "/checkout/library/std/src/sys/pal/hermit/alloc.rs" "/checkout/library/std/src/sys/pal/hermit/thread.rs" "/checkout/library/std/src/sys/pal/hermit/fd.rs" "/checkout/library/std/src/sys/pal/hermit/io.rs" "/checkout/library/std/src/sys/pal/hermit/args.rs" "/checkout/library/std/src/sys/pal/hermit/mod.rs" "/checkout/library/std/src/sys/pal/hermit/env.rs" "/checkout/library/std/src/sys/pal/xous/net/dns.rs" "/checkout/library/std/src/sys/pal/xous/net/udp.rs" "/checkout/library/std/src/sys/pal/xous/net/tcpstream.rs" "/checkout/library/std/src/sys/pal/xous/net/mod.rs" "/checkout/library/std/src/sys/pal/xous/net/tcplistener.rs" "/checkout/library/std/src/sys/pal/xous/os.rs" "/checkout/library/std/src/sys/pal/xous/stdio.rs" "/checkout/library/std/src/sys/pal/xous/time.rs" "/checkout/library/std/src/sys/pal/xous/alloc.rs" "/checkout/library/std/src/sys/pal/xous/thread.rs" "/checkout/library/std/src/sys/pal/xous/mod.rs" "/checkout/library/std/src/sys/pal/windows/handle.rs" "/checkout/library/std/src/sys/pal/windows/os.rs" "/checkout/library/std/src/sys/pal/windows/process/tests.rs" "/checkout/library/std/src/sys/pal/windows/stdio.rs" "/checkout/library/std/src/sys/pal/windows/time.rs" "/checkout/library/std/src/sys/pal/windows/alloc/tests.rs" "/checkout/library/std/src/sys/pal/windows/net.rs" "/checkout/library/std/src/sys/pal/windows/process.rs" "/checkout/library/std/src/sys/pal/windows/compat.rs" "/checkout/library/std/src/sys/pal/windows/futex.rs" "/checkout/library/std/src/sys/pal/windows/stdio/tests.rs" "/checkout/library/std/src/sys/pal/windows/c/windows_targets.rs" "/checkout/library/std/src/sys/pal/windows/c/windows_sys.rs" "/checkout/library/std/src/sys/pal/windows/stack_overflow_uwp.rs" "/checkout/library/std/src/sys/pal/windows/pipe.rs" "/checkout/library/std/src/sys/pal/windows/fs.rs" "/checkout/library/std/src/sys/pal/windows/handle/tests.rs" "/checkout/library/std/src/sys/pal/windows/rand.rs" "/checkout/library/std/src/sys/pal/windows/stack_overflow.rs" "/checkout/library/std/src/sys/pal/windows/alloc.rs" "/checkout/library/std/src/sys/pal/windows/os/tests.rs" "/checkout/library/std/src/sys/pal/windows/thread.rs" "/checkout/library/std/src/sys/pal/unix/alloc.rs"` failed.
If you're running `tidy`, try again with `--bless`. Or, if you just want to format code, run `./x.py fmt` instead.
  local time: Tue Jul 16 22:21:11 UTC 2024
  network time: Tue, 16 Jul 2024 22:21:11 GMT
##[error]Process completed with exit code 1.
Post job cleanup.
```

</details>

--- a/library/std/src/sys/pal/zkvm/alloc.rs
+++ b/library/std/src/sys/pal/zkvm/alloc.rs
@@ -5,7 +5,7 @@ use crate::alloc::{GlobalAlloc, Layout, System};
 unsafe impl GlobalAlloc for System {
     #[inline]
     unsafe fn alloc(&self, layout: Layout) -> *mut u8 {
-        abi::sys_alloc_aligned(layout.size(), layout.align())
+        unsafe { abi::sys_alloc_aligned(layout.size(), layout.align()) }
     }
 
     #[inline]

--- a/library/std/src/sys/pal/zkvm/mod.rs
+++ b/library/std/src/sys/pal/zkvm/mod.rs
@@ -6,6 +6,7 @@
 //! This is all super highly experimental and not actually intended for
 //! wide/production use yet, it's still all in the experimental category. This
 //! will likely change over time.
+#![forbid(unsafe_op_in_unsafe_fn)]
 
 const WORD_SIZE: usize = core::mem::size_of::<u32>();
 

